component/otelcol/receiver/prometheus/prometheus.go

```diff
@@ -0,0 +1,149 @@
+// Package prometheus provides an otelcol.receiver.prometheus component.
+package prometheus
+
+import (
+	"context"
+	"os"
+	"regexp"
+	"sort"
+	"sync"
+	"time"
+
+	"github.com/go-kit/log"
+	"github.com/grafana/agent/component"
+	"github.com/grafana/agent/component/otelcol"
+	"github.com/grafana/agent/component/otelcol/internal/fanoutconsumer"
+	"github.com/grafana/agent/component/otelcol/receiver/prometheus/internal"
+	"github.com/grafana/agent/pkg/build"
+	"github.com/grafana/agent/pkg/util/zapadapter"
+	"github.com/prometheus/prometheus/model/labels"
+	"github.com/prometheus/prometheus/storage"
+	otelcomponent "go.opentelemetry.io/collector/component"
+	otelconfig "go.opentelemetry.io/collector/config"
+	"go.opentelemetry.io/otel/metric"
+	"go.opentelemetry.io/otel/trace"
+)
+
+func init() {
+	component.Register(component.Registration{
+		Name:    "otelcol.receiver.prometheus",
+		Args:    Arguments{},
+		Exports: Exports{},
+
+		Build: func(o component.Options, a component.Arguments) (component.Component, error) {
+			return NewComponent(o, a.(Arguments))
+		},
+	})
+}
+
+// Arguments configures the otelcol.receiver.prometheus component.
+type Arguments struct {
+	ExternalLabels map[string]string `river:"external_labels,attr,optional"`
```

Comment:
I feel a little bit weird about supporting injecting custom labels; it seems like you should be using prometheus.relabel for this. 

Can we take this out of the component for now?

Reply:
Yeah, I'm on it.